tool-metagen/src/DA/Daml/GenDaml.hs

```diff
@@ -8,16 +8,19 @@ import           DA.Daml.TypeModel
 import           Text.PrettyPrint.HughesPJ as PP
 
 import qualified Data.List as List
+import qualified Data.Set as Set
 
 -- | Convert from the TypeModel to a DAML source text.
 ppModule :: Show a => Module a -> Doc
 ppModule Module{..} =
-    text "daml 1.2"
-        $$ text "-- Generated by Metagen"
+    let primitiveTypes = List.foldl' usedPrimitives Set.empty module_decls
```

Comment:
Just to make sure I'm understanding this  - You're explicitly importing all the different types used?

If so, while I agree with this in principle, it's adding a considerable amount of code and complexity. Is it worth reconsidering this?

Reply:
The origin of this request was to restrict how `Prelude` is imported as there are conflicts with things defined in Prelude and a name generated from the Xml. This search restricts that import from Prelude of _just_ the types necessary to compile the module. Yes it is a bit of code. Maybe it would be simpler to just import all 6 of them ( Text, Bool, Int, Decimal, Time, and Date), but we want to precise 😄

Reply:
So if I got this correctly, you're saying that the generated `*.daml` file would have a type like `Sum` or `Logic` defined in there, that also appears in `Prelude`?

I'm not sure that's going to save you; I think a better solution might be using explicit namespaces for the generated files. i.e. `import ... qualified as ...`. That way you won't run into issues if you have e.g. a `Date` or `Time` xml element (not that unlikely, I'd say ...)

I guess my only gripe here is the amount of added code vs functionality. But all good!

Reply:
No, the code just checks if the generated code imports primitives like `Date`, `Time`, `Party` and only imports those from Prelude.